### DISCUSSION TITLE
icons

### DISCUSSION THREAD

**Discussion by NIKOLENKO-TE** [moudey/Shell#554] [Q&A]
Can you make a separate page (like with glyphs) but about icons and methods of working with them, like:
image=**icon.settings**(**auto**, image.**color1**)
image=icon.desktop
image=icon.settings

etc

**Comment by RubicBG** [comment]
- https://nilesoft.org/docs/functions/id - this page is 85% complete (replace `id.` with `icon.`)
- all images (no exceptions) that are written as a function (as you showed) are defined as svg in `images.nss` file 
- there are three types of syntax: 
  - `icon.settings` - as defined in `images.nss` file
  - `icon.settings(#ff0)` - where only the main color changes
  - `icon.settings(color.red, color.green)` - where the main and second color are change
 - words that can be used in brackets
   - `auto` - skip changes
   -  `image.color1`,  `image.color2`,  `image.color3` - NS works with 3-color-system
   - predefined colors and functions https://nilesoft.org/docs/functions/color
   - `#ffff000` or `#ff0` - hex color string
   
  **bonus tip:**
  - `image-sel` is experimental, not documented anywhere and does not work with all syntax - but it works here. It is used together with `image`
  ```js
  image=icon.copy(#ff0, #f00) image-sel=icon.copy(#f00, #ff0)
  image=color.transparent image-sel=icon.delete(auto, color.red)
  ```
